### ISSUE TITLE
Chart Improvements

### ISSUE THREAD

**Issue by anaisconce** [gristlabs/grist-core#81]
Improve Chart UI to present more configuration options; add new chart types such as gauge, donut, radar (aka spider or web), and stacked bar chart.

**Comment by anaisconce** [comment]
Request to add data labels to charts (not just on hover)

**Comment by briantegridy** [comment]
Would be great to choose colors as well.

**Comment by Eisfunke** [comment]
Hi Grist team, it might be worth to consider integrating something like [Vega Lite](https://vega.github.io/vega-lite/). Vega (Lite) is a JSON-based format to specify various graphs. It's very powerful and can output charts and graphs of a lot of types with a lot of customization options. From the code Vega generates interactive HTML graphs from that specification which should integrate well into Grist's web UI.

Integrating something like that, with access to the corresponding spreadsheet's data, would immediately give you access to all kinds of charts and features with a lot of customization options supported by Vega, without implementing them all yourself one-by-one. You would also gain integration with the other features of that ecosystem, e.g. SVG/PNG export. For the more basic features, you could still provide GUI elements, so it shouldn't be an UX downgrade, but more advanced users could code custom charts to their liking.

**Comment by paulfitz** [comment]
@Eisfunke good idea. Reminds me a bit of this custom widget https://github.com/gristlabs/grist-widget/pull/37 to let you do arbitrary plotly charts. (The widget never got landed but a preview is available using custom widget url https://deploy-preview-37--boisterous-sunburst-a5c941.netlify.app/custom-plotly-chart). I think someone could do a Vega Lite version as a custom widget - it would have all the live data immediately, all that is needed is to render it. A full integrated feature would be even nicer, but significantly more work. Notes on how to make custom widgets are here https://support.getgrist.com/widget-custom/

**Comment by anaisconce** [comment]
We now have a widget for advanced charts: https://github.com/gristlabs/custom-charts-widget/blob/main/USAGE.md 📈

**Comment by pierrecamilleri** [comment]
For anyone stumbling upon this issue, we have developed a Vega-Lite widget for advanced visualizations ! (ping @Eisfunke) 

Code source and use instructions : https://gitlab.com/multi-coop/grist-vega-widget/

**Comment by dsagal** [comment]
@pierrecamilleri - the demo looks great!

**Comment by castanearie** [comment]
I wanted a fully offline capable Vega-Lite widget, so I ended up writing my own. It uses a framework, requires a build step, but bundles all dependencies. https://codeberg.org/Castanearie/grist-svega
